health-monitors/kubernetes-object-monitor/pkg/controller/reconciler_test.go

```diff
@@ -0,0 +1,386 @@
+// Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	celenv "github.com/nvidia/nvsentinel/health-monitors/kubernetes-object-monitor/pkg/cel"
+	"github.com/nvidia/nvsentinel/health-monitors/kubernetes-object-monitor/pkg/config"
+	"github.com/nvidia/nvsentinel/health-monitors/kubernetes-object-monitor/pkg/controller"
+	"github.com/nvidia/nvsentinel/health-monitors/kubernetes-object-monitor/pkg/metrics"
+	"github.com/nvidia/nvsentinel/health-monitors/kubernetes-object-monitor/pkg/policy"
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+// To run these tests, you need to install and setup envtest:
+//   go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
+//   source <(setup-envtest use -p env)
+//
+// Then run the tests:
+//   go test -v ./pkg/controller/...
+
+func TestReconciler_NodeHealthyToUnhealthy(t *testing.T) {
```

Comment:
Can we add a test case for cold start: i.e., there were some healthy -> unhealthy transitions and the controller was restarted, post restart it should also ensure:
- if the resource was deleted, we get healthy
- if the resource is still unhealthy, we get unhealthy 
- if the resource has become healthy, we get healthy